sprint_09

SharaldaevBair opened the pull request:
Экран профиля пользователя, свёрстанный кодом.
Кнопка «Поделиться» на экране приложения.

SharaldaevBair committed:
Экран профиля пользователя, свёрстанный кодом.
Кнопка «Поделиться» на экране приложения.

Helvetii951 commented on EndlessTape/EndlessTape/Profile/ProfileViewController.swift:
```diff
@@ -1,13 +1,59 @@
+        let loginNameLabel = UILabel()
+        loginNameLabel.text = "@ekaterina_nov"
+        loginNameLabel.textColor = .gray
+        loginNameLabel.font = UIFont(name: "Regular", size: 13)
```
Шрифт `"Regular"` не был добавлен в проект. Скорее всего имелся ввиду системный шрифт с `.weight` `.regular`. Нужно использовать функцию [`systemFont(ofSize:weight:)`](https://developer.apple.com/documentation/uikit/uifont/1619027-systemfont). Код будет выглядеть так:

`label.font = UIFont.systemFont(ofSize: 13, weight: .regular)` . `weight` в этом случае можно не указывать, потому что он по-умолчанию `.regular`.

Helvetii951 commented on EndlessTape/EndlessTape/SingleImage/SingleImageViewController.swift:
```diff
@@ -5,14 +5,55 @@ final class SingleImageViewController: UIViewController {
     
-    @IBOutlet var imageView: UIImageView!
+    @IBOutlet private var imageView: UIImageView!
+    @IBOutlet weak var scrollView: UIScrollView!
```
Рекомендую сделать `scrollView` приватным.

Helvetii951 commented on EndlessTape/EndlessTape/SingleImage/SingleImageViewController.swift:
```diff
@@ -5,14 +5,55 @@ final class SingleImageViewController: UIViewController {
+        dismiss(animated: true, completion: nil)
+    }
+    
+    @IBAction func didTapShareButton(_ sender: UIButton) {
```
Этот `@IBAction` тоже лучше сделать приватным.

Helvetii951 commented on EndlessTape/EndlessTape/Profile/ProfileViewController.swift:
```diff
@@ -1,13 +1,59 @@
+        
+        let logoutButton = UIButton(type: .system)
+        logoutButton.setImage(UIImage(systemName: "ipad.and.arrow.forward"), for: .normal)
+        logoutButton.tintColor = .red
```
Нужен `YP Red`. Можно либо добавить в проект цвет из макета, либо экспортировать картинку из Figma.

Helvetii951 commented on EndlessTape/EndlessTape/Profile/ProfileViewController.swift:
```diff
@@ -1,13 +1,59 @@
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        let avatarImageView = UIImageView()
+        avatarImageView.image = UIImage(named: "Photo")
+        avatarImageView.translatesAutoresizingMaskIntoConstraints = false
```
Можно вынести элементы верстки(картинку, лэйблы) в свойства контроллера, чтобы к ним был доступ вне кода настройки верстки. Сейчас это не критично. Но в следующих спринтах это будет необходимо.

SharaldaevBair committed:
1. Исправлены шрифты при верстке кодом
2. Исправлен цвет на кнопке logoutButton на экране пользователя, на цвет из макета
3. Элементы верстки кодом в ProfileViewController вынесены в приватные методы
